src/App.tsx

```diff
@@ -1,61 +1,135 @@
 import './App.scss';
+import { useState, FC, ChangeEvent } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './components/UserInfo';
+import { Todo } from './components/TodoInfo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundedUser = usersFromServer.find(user => user.id === userId);
+
+  return foundedUser || null;
+}
+
+export const App: FC = () => {
+  const initialTodos = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
+
+  const [todo, setTodos] = useState<Todo[]>(initialTodos);
+
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [formErrors, setFormErrors]
+    = useState<{ title?: string, user?: string }>({});
+
+  const handleFormSubmit = (event: ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!selectedUserId) {
+      setFormErrors((prevErrors) => ({
+        ...prevErrors,
+        user: 'Please choose a user',
+      }));
+    }
+
+    if (title.trim() === '') {
+      setFormErrors((prevErrors) => ({
+        ...prevErrors,
+        title: 'Please enter a title',
+      }));
+    }
+
+    if (selectedUserId && title.trim() !== '') {
+      const newId
+        = todo.reduce((maxId, item) => Math.max(maxId, item.id), 0) + 1;
+
+      const newUser = getUser(selectedUserId);
+
+      const newTodo: Todo = {
+        id: newId,
+        title: title.trim(),
+        userId: selectedUserId,
+        completed: false,
+        user: newUser,
+      };
+
+      setTodos([...todo, newTodo]);
```

Comment:
good practice to use a callback when we need a prevState
```suggestion
      setTodos((prevTodos) => [...prevTodos, newTodo]);
```